core/includes/database/select.inc

```diff
@@ -1934,12 +1934,12 @@ class SelectQuery extends Query implements SelectQueryInterface {
     $fields = array();
     foreach ($this->tables as $alias => $table) {
       if (!empty($table['all_fields'])) {
-        $fields[] = $this->connection->escapeTable($alias) . '.*';
+        $fields[] = $this->connection->escapeAlias($alias) . '.*';
       }
     }
     foreach ($this->fields as $alias => $field) {
       // Always use the AS keyword for field aliases.
-      $fields[] = (isset($field['table']) ? $this->connection->escapeTable($field['table']) . '.' : '') . $this->connection->escapeField($field['field']) . ' AS ' . $this->connection->escapeAlias($field['alias']);
+      $fields[] = (isset($field['table']) ? $this->connection->escapeAlias($field['table']) . '.' : '') . $this->connection->escapeField($field['field']) . ' AS ' . $this->connection->escapeAlias($field['alias']);
```

Comment:
It seems to me that `escapeTable` should be applied to tables, and `escapeAlias` should be applied to aliases. Does this work?

```suggestion
      $fields[] = (isset($field['table']) ? $this->connection->escapeTable($field['table']) . '.' : '') . $this->connection->escapeField($field['field']) . ' AS ' . $this->connection->escapeAlias($field['alias']);
```

Reply:
@quicksketch I think the fields are defined according to the alias so once you are looking at fields you are already using the table aliases.  If you try what you suggest, you get this error on that view:

> SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'system.`filename` AS `filename` FROM `system` `system` LIMIT 10 OFFSET 0' at line 1

Reply:
For the actual table name, all `escapeTable()` is doing is making sure there are no characters that shouldn't be there.  The function `setPrefix()` does the job of escaping it.  In `/core/includes/database/mysql/database.inc` you can see that `setPrefix()` handles the quotations needed for the actual table name. 

<details>
<summary>/core/includes/database/mysql/database.inc -> setPrefix()</summary>

```php
  /**
   * {@inheritdoc}
   */
  protected function setPrefix($prefix) {
    if (is_array($prefix)) {
      $this->prefixes = $prefix + array('default' => '');
    }
    else {
      $this->prefixes = array('default' => $prefix);
    }

    // This differs from the parent class implementation in that all tables
    // are escaped with back-ticks. First, do table-specific replacements.
    $this->prefixSearch = array();
    $this->prefixReplace = array();
    foreach ($this->prefixes as $key => $val) {
      if ($key != 'default') {
        $this->prefixSearch[] = '{' . $key . '}';
        // Add backticks around the entire table name to avoid MySQL keywords,
        // but also add backticks within the prefix value, in the event
        // different database names are used as a prefix.
        $this->prefixReplace[] = '`' . str_replace('.', '`.`', $val) . $key . '`';
      }
    }
    // Then replace remaining tables with the default prefix.
    $this->prefixSearch[] = '{';
    $this->prefixReplace[] = '`' . $this->prefixes['default'];
    $this->prefixSearch[] = '}';
    $this->prefixReplace[] = '`';
  }
```

</details>

I think trying to add `escapeTable()` into `DatabaseConnection_mysql` with `quoteIdentifier()` in is going to clash with what is happening in `setPrefix()`

Reply:
Yep, @yorkshire-pudding I see the problem now (finally). Everywhere else `escapeTable()` is used it's surrounded by the `{` and `}` braces. For example:

```
'DELETE FROM {' . $this->connection->escapeTable($this->table) . '} '
```

Exaborating on why you're right: If we _really_ escape the table by adding back-ticks here, it would break table prefixes. i.e. a table prefix of `example_` and a table `node` would result in a query like ``DELETE FROM example_`node` `` (incorrect). What is should be is  ``DELETE FROM `example_node` ``, with the backticks around both the table name and prefix.

So `escapeTable()` doesn't do the job I expected, and if we added backticks to that method it would break a bunch of other stuff. So your fix looks correct and the best available. Thank you for your patience!